### PR TITLE
Intercom triggers

### PR DESCRIPTION
**Pull request by phoenixeliot**
Reports some new intercom-related values to Intercom and makes sure the user gets the intercom bubble opened when they happen.

### CODE REVIEW

File: app/schemas/models/user.coffee [hunk 1]
```diff
@@ -379,6 +379,13 @@ _.extend UserSchema.properties,
   lastAnnouncementSeen:
     type: 'number'
     description: 'The highed announcement modal index displayed to the user.'
+  intercomTriggers:
+    type: 'object'
+    description: 'Flags for whether certain intercom triggers have been activated'
+    properties: {
+      studentStartedWakkaMaul: { type: 'boolean', description: "One of a teacher's students has reached Wakka Maul" }
+      studentFinishedMasterOfNames: { type: 'boolean', description: "One of a teacher's students has finished Master of Names" }
+    }
```
Comment (sderickson) [inline app/schemas/models/user.coffee]: I'm tempted to have these in `activity` instead. Seems a bit off to have a property object specific for intercom. Also, wasn't `studentFinishedMasterOfNames` changed to starting the next level?

File: app/core/application.coffee [hunk 2]
```diff
@@ -76,6 +76,10 @@ Application = {
     $(document).bind 'keydown', preventBackspace
     preload(COMMON_FILES)
     CocoModel.pollAchievements()
+    if me.isTeacher() and @isProduction() and not @testing
+      setInterval =>
+        window.tracker.identify() unless @userIsIdle
```
Comment (sderickson) [inline app/core/application.coffee]: Change to `Intercom('update')`.

File: server/middleware/levels.coffee [hunk 3]
```diff
@@ -126,5 +128,38 @@ module.exports =
 
     attrs.isForClassroom = course?
     session = new LevelSession(attrs)
+    if classroom # Potentialy set intercom trigger flag on teacher
```
Comment (sderickson) [inline server/middleware/levels.coffee]: Potentially

File: server/middleware/levels.coffee [hunk 4]
```diff
@@ -126,5 +128,38 @@ module.exports =
...
+  if update
+    tries = 0
+    while tries < 100
+      tries += 1
+      try
+        console.log yield intercom.users.create update
+        break
+      catch e
+        console.log "couldn't update intercom", e
+        yield new Promise (accept, reject) -> setTimeout(accept, 5000)
```
Comment (sderickson) [inline server/middleware/levels.coffee]: Would be good to test this function, especially the retry functionality. Does the call often fail?

Comment (phoenixeliot) [inline server/middleware/levels.coffee]: It uses the same API as some of the sales automation scripts, which are designed to run as close to capacity as possible.

**Pull request merged**